BrawlLib/Wii/Animations/CHR0SMDImporter.cs

```diff
@@ -0,0 +1,175 @@
+                        float rotY = float.Parse(frameNumbers[7], CultureInfo.InvariantCulture.NumberFormat) * radianFloat;
+                        float rotZ = float.Parse(frameNumbers[8], CultureInfo.InvariantCulture.NumberFormat) * radianFloat;
+
+                        currentFrame = new SMDFrame(currentFrameID, boneID, posX, posY, posZ, rotX, rotY, rotZ);
+
+                        allFrames.Add(currentFrame);
+                    }
+
+                    if (lineNumber == 0) //First Line
+                    {
+                        if (line.CompareTo("version 1") != 0) //The first line contains the version number. It needs to be 1.
+                        {
+                            MessageBox.Show("Unsupported SMD Version. Please use SMD Version 1."); //I don't even know if there's versions >1 but let's not take risks
+                            return null;
+                        }
+                    }
+
+                    if (line.CompareTo("nodes") == 0) //When reaching this line, we start reading bones names
+                    {
+                        isReadingBones = true;
+                    }
+
+                    if (line.CompareTo("skeleton") == 0) //When reaching this line, we start reading frames
+                    {
+                        isReadingFrames = true;
+                    }
+
+                    lineNumber++;
+                }
+
+                foreach (int thisBone in boneIDs)
+                {
+                    CHR0EntryNode node = new CHR0EntryNode();
+                    chr0.AddChild(node);
+                    node.Name = boneNames[thisBone];
+                    node.SetSize(currentFrameID + 1, true);
+
+                    int keynum = 0;
+
+                    foreach (SMDFrame currentFrame in allFrames)
+                    {
+                        //Keyframe Array documentation:
+                        //  ---------------------
+                        //  |       | X | Y | Z |
+                        //  | Scale | 0 | 1 | 2 |
+                        //  | Rot   | 3 | 4 | 5 |
+                        //  | Pos   | 6 | 7 | 8 |
+                        //  ---------------------
+                        if (currentFrame.boneID == thisBone)
+                        {
+                            node.SetKeyframe(3, currentFrame.id, currentFrame.rotx);
+                            node.SetKeyframe(4, currentFrame.id, currentFrame.roty);
+                            node.SetKeyframe(5, currentFrame.id, currentFrame.rotz);
+                            node.SetKeyframe(6, currentFrame.id, currentFrame.posx);
+                            node.SetKeyframe(7, currentFrame.id, currentFrame.posy);
+                            node.SetKeyframe(8, currentFrame.id, currentFrame.posz);
+
+                            keynum++;
+                        }
+                    }
+
+                    chr0.FrameCount = keynum;
+                    chr0.Loop = true;
```

Comment:
This should not be set repeatedly. Set it in the declaration (you currently set it to false there)